src/niri.rs

```diff
@@ -1168,10 +1169,15 @@ impl Niri {
 
                     if let Some(mut overlap) = geo.intersection(bbox) {
                         overlap.loc -= surface_pos;
+                        dnd_scale = dnd_scale.max(output.current_scale().integer_scale());
                         output_update(output, Some(overlap), surface);
                     } else {
                         output_update(output, None, surface);
                     }
+
+                    with_states(surface, |data| {
+                        send_surface_state(surface, data, dnd_scale, Transform::Normal);
```

Comment:
```suggestion
                        send_surface_state(surface, data, dnd_scale, output.current_transform());
```

Reply:
There's no `output` to call `current_transform`, since the surface can span multiple outputs and deciding which transform to use is not as trivial as `scale` where you can pick max.

So you need some heuristics to pick transform the way we pick scale and then apply, what you've suggested doesn't work.

Reply:
Right, I see.